test/index.cpp

```diff
@@ -709,6 +709,15 @@ TEST(lookup, largeDim)
     af::array output = af::lookup(input, indices);
 }
 
+TEST(lookup, Issue2009)
+{
+    af::array a   = af::range(af::dim4(1000, 1));
+    af::array idx = af::constant(0, 1, u32);
+    af::array b   = af::lookup(a, idx, 1);
+
+    ASSERT_EQ(true, af::allTrue<bool>(a==b));
```

Comment:
We need to figure out a way to print values that are not correct when a test fails so that we don't have to dig into the functions to figure out what is wrong. If this test were to fail we wouldn't know what characteristics the array has when it fails.

Reply:
I think we can generalize couple of helper functions from all the tests we currently have and use those everywhere. In those functions, whenever the test fails, we print out the the details to log.

Reply:
stdout is fine. gtest has ways to print custom objects. We need to start thinking about doing that because we waste a lot of time figuring out how something is failing.

Reply:
Yeah, stdout is good, by log I meant the ci logs. True.